inc/entity.class.php

```diff
@@ -512,6 +547,68 @@ public static function cronCreditExpired($task)
         return 1;
     }
 
+    public static function cronLowCredits($task)
+    {
+        /**
+         * @var array $CFG_GLPI
+         * @var DBmysql $DB
+         */
+        global $CFG_GLPI, $DB;
+
+        if (!$CFG_GLPI['use_notifications']) {
+            return 0;
+        }
+
+        $alert = new Alert();
+        $credits_iterator = $DB->request(
+            [
+                'SELECT' => [
+                    'glpi_plugin_credit_entities.id',
+                    'glpi_plugin_credit_entities.name',
+                ],
+                'FROM' => 'glpi_plugin_credit_entities',
+                'LEFT JOIN' => [
+                    'glpi_plugin_credit_tickets' => [
+                        'ON' => [
+                            'glpi_plugin_credit_tickets' => 'plugin_credit_entities_id',
+                            'glpi_plugin_credit_entities' => 'id',
+                        ]
+                    ]
+                ],
+                'WHERE' => [
+                    'glpi_plugin_credit_entities.is_active' => 1,
+                ],
+                'GROUPBY' => 'glpi_plugin_credit_entities.id',
+                'HAVING' => new QueryExpression('glpi_plugin_credit_entities.quantity - quantity_consumed <= (glpi_plugin_credit_entities.quantity * glpi_plugin_credit_entities.low_credit_alert) / 100')
+            ]
+        );
+
+        foreach ($credits_iterator as $credit_data) {
+            $task->addVolume(1);
+            $task->log(
+                sprintf(
+                    'Credit %s has been consumed',
+                    $credit_data['name'],
+                )
+            );
```

Comment:
Is this log correct ? I am not sure we are consuming credits here.

Reply:
what I meant is that the threshold has been exceeded so yes maybe "low credit for %s" is more appropriate

Reply:
Yes, it would be more appropriate 👍